packages/flutter/lib/src/foundation/diagnostics.dart

```diff
@@ -1605,49 +1605,64 @@ abstract class DiagnosticsNode {
   ///    by this method and interactive tree views in the Flutter IntelliJ
   ///    plugin.
   @mustCallSuper
-  Map<String, Object?> toJsonMap(DiagnosticsSerializationDelegate delegate) {
+  Map<String, Object?> toJsonMap(
+    DiagnosticsSerializationDelegate delegate, {
+    bool fullDetails = true,
+  }) {
     Map<String, Object?> result = <String, Object?>{};
     assert(() {
       final bool hasChildren = getChildren().isNotEmpty;
-      result = <String, Object?>{
+      final Map<String, Object?> essentialDetails = <String, Object?>{
         'description': toDescription(),
-        'type': runtimeType.toString(),
-        if (name != null)
-          'name': name,
-        if (!showSeparator)
-          'showSeparator': showSeparator,
-        if (level != DiagnosticLevel.info)
-          'level': level.name,
-        if (!showName)
-          'showName': showName,
-        if (emptyBodyDescription != null)
-          'emptyBodyDescription': emptyBodyDescription,
-        if (style != DiagnosticsTreeStyle.sparse)
-          'style': style!.name,
-        if (allowTruncate)
-          'allowTruncate': allowTruncate,
-        if (hasChildren)
-          'hasChildren': hasChildren,
-        if (linePrefix?.isNotEmpty ?? false)
-          'linePrefix': linePrefix,
-        if (!allowWrap)
-          'allowWrap': allowWrap,
-        if (allowNameWrap)
-          'allowNameWrap': allowNameWrap,
-        ...delegate.additionalNodeProperties(this),
-        if (delegate.includeProperties)
-          'properties': toJsonList(
-            delegate.filterProperties(getProperties(), this),
-            this,
-            delegate,
-          ),
+        'shouldIndent': style != DiagnosticsTreeStyle.flat &&
+            style != DiagnosticsTreeStyle.error,
+        ...delegate.additionalNodeProperties(this, fullDetails: fullDetails),
         if (delegate.subtreeDepth > 0)
           'children': toJsonList(
             delegate.filterChildren(getChildren(), this),
             this,
             delegate,
+            fullDetails: fullDetails,
           ),
       };
+
+      if (!fullDetails) {
+        result = essentialDetails;
+      } else {
+        result = <String, Object?>{
```

Comment:
nit: this could be more terse with a ternary
```
result = fullDetails
  ? { ... }
  : essentialDetails;
```

Reply:
Done!